logic/overworld.py

```diff
@@ -250,7 +262,9 @@ def __init__(self, options, world_setup, r):
         Location().add(KeyLocation("CASTLE_BUTTON")).connect(castle_inside, None)
         castle_top_outside = Location()
         castle_top_inside = Location()
-        self._addEntrance("castle_main_entrance", castle_frontdoor, castle_inside, r.bush)
```

Comment:
Seems like castle_frontdoor is the same node as the new outside_castle_main_entrance, this piece can be cleaned up to use castle_frontdoor instead of outside_castle_main_entrance, unless you had a specific reason for this structure? 

This version requires a bush breaker to leave the castle if the gate is open, which shouldn't be the case.

Reply:
Oh, good catch. I was going through all entrances and adding a new node when I found one I couldn't go back through, I didn't look at this case close enough. Should be fixed now.